client/analytics/report/orders/index.js

```diff
@@ -0,0 +1,81 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, Fragment, compose } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import { withSelect, withDispatch } from '@wordpress/data';
+import moment from 'moment';
+import { partial } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Header from 'layout/header/index';
+import Card from 'components/card';
+import { requestUpdateOrder } from './store/mutations';
+import './store';
+
+class OrdersReport extends Component {
+	render() {
+		const { orders, onToggleStatus } = this.props;
+		return (
+			<Fragment>
+				<Header
+					sections={ [
+						[ '/analytics', __( 'Analytics', 'woo-dash' ) ],
+						__( 'Orders', 'woo-dash' ),
+					] }
+				/>
+				<Card title="Orders">
+					<table style={ { width: '100%' } }>
+						<thead>
+							<tr style={ { textAlign: 'left' } }>
+								<th>Id</th>
+								<th>Date</th>
+								<th>Total</th>
+								<th>Status</th>
+								<th>Action</th>
+							</tr>
+						</thead>
+						<tbody>
+							{ orders &&
+								orders.map( order => (
+									<tr>
+										<td>{ order.id }</td>
+										<td>{ moment( order.date_created ).format( 'LL' ) }</td>
+										<td>{ order.total }</td>
+										<td>{ order.status }</td>
+										<td>
+											<Button isPrimary onClick={ partial( onToggleStatus, order ) }>
+												Toggle Status
+											</Button>
+										</td>
+									</tr>
+								) ) }
+						</tbody>
+					</table>
+				</Card>
+			</Fragment>
+		);
+	}
+}
+
+export default compose(
+	withSelect( select => {
+		const { getOrders } = select( 'woo-dash' );
+		return {
+			orders: getOrders(),
+		};
+	} ),
+	withDispatch( dispatch => {
+		return {
```

Comment:
@aduth is this where you see the resolvers at work?

this is where I see `requestHttpData()` coming into play since it can trigger side-effects and through Redux update state on those transitions without breaking from the synchronous paradigm or resorting to thunks